annotation/app/tasks/resources.py

```diff
@@ -949,6 +957,91 @@ def finish_task(
                 detail=f"Error: connection error ({exc.exc_info})",
             )
         update_inner_job_status(db, task.job_id, JobStatusEnumSchema.finished)
+
+    elif (
+        job.validation_type == ValidationSchema.extensive_coverage
+        and os.getenv("AGREEMENT_SCORE_ENABLED", "false") == "true"
+    ):
+        annotation_tasks: List[ManualAnnotationTask] = (
+            db.query(ManualAnnotationTask)
+            .filter(
+                ManualAnnotationTask.job_id == task.job_id,
+                ManualAnnotationTask.file_id == task.file_id,
+                ManualAnnotationTask.is_validation.is_(False),
+            )
+            .count()
+        )
```

Comment:
May you move this DB interaction to `services.py` to separate this logic from controller?